boards/common/esp32/Kconfig

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2020 HAW Hamburg
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+config BOARD_COMMON_ESP32
+    bool
+    select HAS_PERIPH_GPIO
+    select HAS_PERIPH_GPIO_IRQ
+    select HAS_PERIPH_UART
+    select HAS_PERIPH_UART_MODECFG
+    select HAS_ESP_SPIFFS
+    select HAS_ESP_WIFI
+    select HAS_ESP_NOW
```

Comment:
So these would be provided by `CPU_COMMON_ESP`? In that case we would depend on #14117

Reply:
Exactly. PR #14117 is close to being merged. Since PR No. 14117 introduces a new ESP WiFi netdev device, we take the opportunity to clean up the feature definition a bit for consistency reasons.

But I suppose you can change it in this PR independently from PR #14117. If features defined in `Kconfig` differ from those defined in `Makefile.features`, this doesn't currently affect the compilation, right?

Reply:
The test would not pass, as those features would be provided for esp8266 in Kconfig, while in Makefile they are not

Reply:
Hm, but this feature was not previously provided for ESP8266 board or CPU makefiles, which is an inconsistency.

Reply:
That's my point. Either we place the features in `CPU_COMMON_ESP` (which means that esp8266 and esp32 will provide them) and wait for #14117, or we leave them here (only esp32 will provide them) and #14117 edits the Kconfig when it's ready to be merged.